var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -170,140 +231,200 @@ def check_darwin_lldb_codesign_requirement(self):
-        if '+lld' in spec:
-            projects.append('lld')
-        if '+compiler-rt' in spec:
-            projects.append('compiler-rt')
-        if '+libcxx' in spec:
-            projects.append('libcxx')
-            projects.append('libcxxabi')
-            if spec.satisfies('@3.9.0:'):
-                cmake_args.append('-DCLANG_DEFAULT_CXX_STDLIB=libc++')
-        if '+internal_unwind' in spec:
-            projects.append('libunwind')
-        if '+polly' in spec:
-            projects.append('polly')
-            cmake_args.append('-DLINK_POLLY_INTO_TOOLS:Bool=ON')
-
-        if '+shared_libs' in spec:
-            cmake_args.append('-DBUILD_SHARED_LIBS:Bool=ON')
-
-        if '+link_dylib' in spec:
-            cmake_args.append('-DLLVM_LINK_LLVM_DYLIB:Bool=ON')
-
-        if '+all_targets' not in spec:  # all is default on cmake
-
-            targets = ['NVPTX', 'AMDGPU']
-            if (spec.version < Version('3.9.0')):
+        if "+cuda" in spec:
+            cmake_args.extend(
+                [
+                    "-DCUDA_TOOLKIT_ROOT_DIR:PATH=" + spec["cuda"].prefix,
+                    "-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES={0}".format(
+                        ",".join(spec.variants["nvptx_offload_ccs"].value)
+                    ),
+                    "-DCLANG_OPENMP_NVPTX_DEFAULT_ARCH=sm_{0}".format(
+                        spec.variants["nvptx_offload_ccs"].value[-1]
+                    ),
+                ]
+            )
+        else:
+            # still build libomptarget but disable cuda
+            cmake_args.extend(
+                [
+                    "-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE",
+                    "-DCUDA_SDK_ROOT_DIR:PATH=IGNORE",
+                    "-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE",
+                    "-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES:STRING=IGNORE",
+                ]
+            )
+
+        if "+omp_debug" in spec:
+            cmake_args.append("-DLIBOMPTARGET_ENABLE_DEBUG:Bool=ON")
+
+        if "+python" in spec and "+lldb" in spec and spec.satisfies("@5.0.0:"):
+            cmake_args.append("-DLLDB_USE_SYSTEM_SIX:Bool=TRUE")
+
+        if "~python" in spec and "+lldb" in spec:
+            cmake_args.append("-DLLDB_DISABLE_PYTHON:Bool=TRUE")
+
+        if "+gold" in spec:
+            cmake_args.append(
+                "-DLLVM_BINUTILS_INCDIR=" + spec["binutils"].prefix.include
+            )
+
+        if "+clang" in spec:
+            projects.append("clang")
+            projects.append("clang-tools-extra")
+            projects.append("openmp")
+        if "+lldb" in spec:
+            projects.append("lldb")
+        if "+lld" in spec:
+            projects.append("lld")
+        if "+compiler-rt" in spec:
+            projects.append("compiler-rt")
+        if "+libcxx" in spec:
+            projects.append("libcxx")
+            projects.append("libcxxabi")
+            if spec.satisfies("@3.9.0:") and '+libcxx_default' in spec:
```

Comment:
Note that this looks like it will break `sollve`, which depends on `libcxx` -- but the semantics have now changed and it won't get the default library it expects.

Reply:
At the moment the sollve package doesn't depend on this one, so that should be ok, but after looking over the history of this bit of code I'll revert it back to removing the libcxx_default option and using it by default if it's built. 
 It seems there was a library search issue with the built clang in some configurations having trouble finding libstdc++.  I'm not sure I'm completely onboard with using libc++ by default, but it looks like it's the lesser of two evils at the moment.